src/reclaim_python_sdk/reclaim.py

```
@@ -321,50 +345,358 @@ def set_app_callback_url(self, url: str, json_proof_response: bool = False) -> N
+            while True:
+                try:
+                    status_url_response = await fetch_status_url(self._session_id)
+                    if not status_url_response.session:
+                        await asyncio.sleep(session_update_polling_interval)
+                        continue
+
+                    status_v2 = status_url_response.session.statusV2
+
+                    if status_v2 == SessionStatus.PROOF_GENERATION_FAILED:
+                        # ignoring complicated timeout logic for python since we just poll endlessly
+                        await asyncio.sleep(session_update_polling_interval)
```

Ruff (0.15.5)

[error] Undefined name `asyncio`

(F821)